version: '3.5'
services:
  rabbitmq-server:
    image: rabbitmq:3.7-management
    hostname: rabbitmq-server
    ports:
      - "15672:15672"
      - "5672:5672"
    healthcheck:
        test: ["CMD", "curl", "-f", "http://localhost:15672"]
        interval: 30s
        timeout: 10s
        retries: 5
    volumes:
      - "/opt/rabbitmq:/var/lib/rabbitmq"
  node-service:
    build: ./node-service
    restart: on-failure
    volumes:
      - ./node-service:/node-app
    links:
      - rabbitmq-server
    depends_on:
      - rabbitmq-server
    tty: true
    ports:
      - "3000:3000"
  python-service:
    build: ./python-service
    restart: on-failure
    volumes:
      - ./python-service:/python-app
    depends_on:
      - rabbitmq-server
    links:
        - rabbitmq-server
    tty: true